src/main/java/parser/ArithmetikParserClass.java

```diff
@@ -0,0 +1,879 @@
+	rightTerm -> div operator rightTerm 
+	rightTerm -> Epsilon
+	operator -> openPar expression closePar | num | ident
+
+
+	Epsilon steht hier f�r das "leere Wort"
+	
+	Der Parser verarbeitet den Token Strom, der durch den Scanner bei der
+	lexikalischen Analyse der Eingabe erzeugt wurde.
+	
+	Der Parser ist nach dem Prinzip des rekursiven Abstiegs programmiert,
+	d.h. jedes nicht terminale Symbol der Grammatik wird durch eine 
+	Methode in Java repr�sentiert, die die jeweils anderen nicht terminalen
+	Symbole auf der rechten Seite der Grammatik Regeln ggf. auch rekursiv
+	aufruft.
+	
+	Der zu parsende Ausdruck wird aus einer Datei gelesen und in einem
+	Array of Char abgespeichert. Pointer zeigt beim Parsen auf den aktuellen
+	Eingabewert.
+	
+	Ist der zu parsende Ausdruck syntaktisch nicht korrekt, so werden 
+	�ber die Methode syntaxError() entsprechende Fehlermeldungen ausgegeben.
+	
+	Zus�tzlich werden den Methoden der Klasse neben der Rekursionstiefe auch
+	eine Referenz auf eine Instanz der Klasse SyntaxTree �bergeben.
+	
+	�ber die Instanzen der Klasse SyntaxTree wird beim rekursiven Abstieg
+	eine konkreter Syntaxbaum des geparsten Ausdrucks aufgebaut.
+
+*/
+
+public class ArithmetikParserClass extends NumScanner implements TokenList{
+	// Konstante f�r Ende der Eingabe
+	public final char EOF=(char)255;
+	// Zeiger auf das aktuelle Eingabezeichen
+	private int pointer;
+	// Zeiger auf das Ende der Eingabe
+	private int maxPointer;
+	// Eingabe zeichenweise abgelegt
+	 private char input[];
+	// Syntaxbaum
+	private SyntaxTree parseTree;
+	
+	//-------------------------------------------------------------------------
+	//------------Konstruktor der Klasse ArithmetikParserClass-----------------
+	//-------------------------------------------------------------------------
+	
+	ArithmetikParserClass(SyntaxTree parseTree){
+		this.parseTree=parseTree;
+		//this.input = new char[256];
+		this.pointer=0;
+		this.maxPointer=0;
+	}
+	
+	ArithmetikParserClass(SyntaxTree parseTree, char[] input) {
+		this.parseTree = parseTree;
+		this.input = input;
+		this.pointer = 0;
+		this.maxPointer = input.length - 1;
+	}
+	
+	//-------------------------------------------------------------------------
+	//-------------------Methoden der Grammatik--------------------------------
+	//-------------------------------------------------------------------------
+	
+	//-------------------------------------------------------------------------
+	// program -> emojiStartCode sequence emojiEndCode
+	// Der Parameter sT ist die Wurzel des bis hier geparsten Syntaxbaumes
+	//-------------------------------------------------------------------------
+	boolean program(SyntaxTree sT){
+		byte [] emojiStartCodeSet = {EMOJI_START_CODE};
+		byte [] emojiEndCodeSet = {EMOJI_END_CODE};
+		byte [] identSet = {IDENT};
+
+		// Falls aktuelle Eingabe ':triangular_flag_on_post:'
+		if (match(emojiStartCodeSet,sT)) {
+    		if (sequence(sT.insertSubtree(SEQUENCE))){
```

Comment:
Die if Bedingung in eine eigene Methode auslagern und mit Titel was sie überprüft versehen.
"public boolean sequenceMatchesBlaBla" -> das führt dazu das der Code wie ein Buch gelesen werden kann.

"if(sequenceMatchesBlaBla) do createSubtree() .."

Reply:
Sind tatsächlich auch ein wenig viele if Blöcke, aber da werde ich jetzt nicht zu sehr in die Tiefe gehen und den Fokus auf das leichter zu fixerende legen.